cairo-vm-cli/src/main.rs

```diff
@@ -160,6 +172,30 @@ fn run(args: impl Iterator<Item = String>) -> Result<(), Error> {
         memory_writer.flush()?;
     }
 
+    if let Some(file_path) = args.air_public_input {
+        let dyn_layout = match args.layout.as_str() {
+            "dynamic" => Some(cairo_runner.get_layout()),
+            _ => None,
+        };
+
+        let public_input = PublicInput::new(
+            cairo_runner.relocated_memory.clone(),
```

Comment:
Could we take a reference instead of cloning the vec?

Reply:
sure! done